src/decanter/core/core_api/predict_input.py

```diff
@@ -49,21 +51,39 @@ def __init__(
         self.pred_body = CoreBody.PredictBody.create(
             data_id='tmp_data_id', model_id='tmp_model_id', callback=callback,
             keep_columns=keep_columns, threshold=threshold, version=version)
+        self.select_by = select_by
+        self.select_model_by = select_model_by
+        self.select_model_id = select_model_id
 
     def getPredictParams(self):
         """Using pred_body to create the JSON request body for prediction.
 
         Returns:
             :obj:`dict`
         """
-        setattr(self.pred_body, 'data_id', self.data.id)
-        setattr(self.pred_body, 'model_id', self.experiment.best_model.id)
+        class_ = self.__class__.__name__
+        if self.select_by == 'by_CVbest':
+            setattr(self.pred_body, 'data_id', self.data.id)
```

Comment:
this line is recurring.

Reply:
sorry, I may not understand the meaning of recursion here,  This recursive situation still appears in the new PR?